go/glide/lib.h

```diff
@@ -0,0 +1,144 @@
+#include <stdarg.h>
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdlib.h>
+
+enum ErrorType {
+  ClosingError = 0,
+  RequestError = 1,
+  TimeoutError = 2,
+  ExecAbortError = 3,
+  ConnectionError = 4,
+};
+typedef uint32_t ErrorType;
+
+enum RequestType {
+  CustomCommand = 1,
+  GetString = 2,
+  SetString = 3,
+  Ping = 4,
+  Info = 5,
+  Del = 6,
+  Select = 7,
+  ConfigGet = 8,
+  ConfigSet = 9,
+  ConfigResetStat = 10,
+  ConfigRewrite = 11,
+  ClientGetName = 12,
+  ClientGetRedir = 13,
+  ClientId = 14,
+  ClientInfo = 15,
+  ClientKill = 16,
+  ClientList = 17,
+  ClientNoEvict = 18,
+  ClientNoTouch = 19,
+  ClientPause = 20,
+  ClientReply = 21,
+  ClientSetInfo = 22,
+  ClientSetName = 23,
+  ClientUnblock = 24,
+  ClientUnpause = 25,
+  Expire = 26,
+  HashSet = 27,
+  HashGet = 28,
+  HashDel = 29,
+  HashExists = 30,
+  MGet = 31,
+  MSet = 32,
+  Incr = 33,
+  IncrBy = 34,
+  Decr = 35,
+  IncrByFloat = 36,
+  DecrBy = 37,
+  HashGetAll = 38,
+  HashMSet = 39,
+  HashMGet = 40,
+  HashIncrBy = 41,
+  HashIncrByFloat = 42,
+  LPush = 43,
+  LPop = 44,
+  RPush = 45,
+  RPop = 46,
+  LLen = 47,
+  LRem = 48,
+  LRange = 49,
+  LTrim = 50,
+  SAdd = 51,
+  SRem = 52,
+  SMembers = 53,
+  SCard = 54,
+  PExpireAt = 55,
+  PExpire = 56,
+  ExpireAt = 57,
+  Exists = 58,
+  Unlink = 59,
+  TTL = 60,
+  Zadd = 61,
+  Zrem = 62,
+  Zrange = 63,
+  Zcard = 64,
+  Zcount = 65,
+  ZIncrBy = 66,
+  ZScore = 67,
+  Type = 68,
+  HLen = 69,
+  Echo = 70,
+  ZPopMin = 71,
+  Strlen = 72,
+  Lindex = 73,
+  ZPopMax = 74,
+  XRead = 75,
+  XAdd = 76,
+  XReadGroup = 77,
+  XAck = 78,
+  XTrim = 79,
+  XGroupCreate = 80,
+  XGroupDestroy = 81,
+};
+typedef uint32_t RequestType;
+
+typedef struct Level Level;
```

Comment:
```suggestion
typedef struct _Level Level;
```
What is `_Level` BTW?

Reply:
Logging level copied from your PR